File: src/main/java/top/xiyang6666/etched_extension/Network.java
package top.xiyang6666.etched_extension;

import net.minecraft.resources.ResourceLocation;
import net.minecraftforge.network.NetworkRegistry;
import net.minecraftforge.network.simple.SimpleChannel;
import top.xiyang6666.etched_extension.packet.EBNRApiPacket;

public class Network {
    public static final String PROTOCOL_VERSION = "1";
    public static final SimpleChannel CHANNEL = NetworkRegistry.newSimpleChannel(
            new ResourceLocation(EtchedExtension.MODID, "main"),
            () -> PROTOCOL_VERSION,
            PROTOCOL_VERSION::equals,
            PROTOCOL_VERSION::equals
    );

    private static int packetId = 0;

    public static void register() {
        CHANNEL.registerMessage(
                packetId++,
                EBNRApiPacket.class,
                EBNRApiPacket::encode,
                EBNRApiPacket::new,
                EBNRApiPacket::handle
        );
    }
}

File: src/main/kotlin/top/xiyang6666/etched_extension/Network.kt
package top.xiyang6666.etched_extension

import net.neoforged.bus.api.SubscribeEvent
import net.neoforged.neoforge.network.event.RegisterPayloadHandlersEvent
import top.xiyang6666.etched_extension.packet.EBNRApiPacket

object Network {
    @SubscribeEvent
    fun register(event: RegisterPayloadHandlersEvent) {
        val registrar = event.registrar("1")
        registrar.playToClient(
            EBNRApiPacket.TYPE,
            EBNRApiPacket.STREAM_CODEC,
            EBNRApiPacket::handle
        )
    }
}